### PR TITLE
Convert to megaparsec

### PR DESCRIPTION
**Pull request by mattoxb**
Here is Megaparsec conversion of the parser.  The test cases that test error messages will still need to be updated, but all the other tests seem to pass now.

### CODE REVIEW

**Comment by traiansf** [comment]
Jenkins: add to whitelist

File: src/main/haskell/language-kore/src/Data/Kore/Parser/LexemeImpl.hs [hunk 1]
```diff
@@ -37,8 +40,9 @@ import qualified Data.ByteString.Char8        as Char8
 import           Data.Char                    (isHexDigit, isOctDigit)
 import           Data.Maybe                   (isJust)
 import qualified Data.Trie                    as Trie
-import qualified Text.Parsec.Char             as Parser (char, string)
-import           Text.Parsec.String           (Parser)
+import           Text.Megaparsec              (takeWhileP)
+import qualified Text.Megaparsec.Char.Lexer   as L
```
Comment (virgil-serbanuta) [inline src/main/haskell/language-kore/src/Data/Kore/Parser/LexemeImpl.hs]: Could you call this 'Lexer'? I think it will make the code easier to follow.

**Comment by traiansf** [comment]
Apparently the comments parser needs to be tuned to fully support C-style comments:
see `src/test/resources/test-comment-5.kore` and several unit tests, including https://github.com/kframework/kore/blob/master/src/main/haskell/language-kore/test/parser/Data/Kore/Parser/LexemeTest.hs#L50

**Comment by bmmoore** [comment]
Rebased and tests fixed in #92

**Pull request closed without merging**